syn/python/get_kge.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/python3
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Parse a yosys area report and give a kGE equivalient
+
+import re
+import argparse
+
+weighted_dict = {}
+
+
+def read_lib(lib_file_path):
+    with open(lib_file_path, 'r') as f:
+        lib_file = f.readlines()
+
+    cell_dict = {}
+    cell_name = ''
+    for line in lib_file:
+        if re.search(r'^  cell \(', line):
+            if cell_name != '':
+                print("Size missing")
+                quit()
+            cell_name = line.split()[1].strip('()')
+        elif re.search(r'^\s+area', line):
```

Comment:
Probably use `re.match` here.

Reply:
Since the other one is matching a literal string, I'll just switch this over too. Weirdly these lines all start with a tab, whereas the cell lines all start with two spaces?